tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,360 @@
+
+fig = plt.figure(constrained_layout=True)
+subfigs = fig.subfigures(1, 2, wspace=0.07, width_ratios=[1.5, 1.])
+axs0 = subfigs[0].subplots(2, 2)
+subfigs[0].set_facecolor('0.9')
+subfigs[0].suptitle('subfigs[0]\nLeft side')
+subfigs[0].supxlabel('xlabel for subfigs[0]')
+
+axs1 = subfigs[1].subplots(3, 1)
+subfigs[1].suptitle('subfigs[1]')
+subfigs[1].supylabel('ylabel for subfigs[1]')
+
+############################################################################
+# It is also possible to nest Axes using `~.pyplot.subplot_mosaic` using
+# nested lists.  This method does not use subfigures, like above, so lacks
+# the ability to add per-subfigure ``suptitle`` and ``supxlabel``, etc.
+# Rather it is a conveneince wrapper around the `~.SubplotSpec.subgridspec`
+# method described below.
+
+inner = [['innerA'],
+         ['innerB']]
+outer = [['upleft',  inner],
+          ['lowleft', 'lowright']]
+
+fig, axd = plt.subplot_mosaic(outer, constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]')
+
+############################################################################
+# Low-level and advanced grid methods
+# ===================================
+#
+# `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` provide high-level
+# interfaces to create `~.gridspec.GridSpec` objects and add associated
+# new Axes to a figure at given `~.gridspec.SubplotSpec` inside the
+# *GridSpec*.
+#
+# Basic 2x2 grid
+# --------------
+#
+# We can accopmplish a 2x2 grid in the same manner as
+# ``plt.subplots(2, 2)``:
+
+fig = plt.figure(figsize=(4.5, 3.5), constrained_layout=True)
+spec = fig.add_gridspec(ncols=2, nrows=2)
+ax0 = fig.add_subplot(spec[0, 0])
+annotate_axes(ax0, 'ax0')
+ax1 = fig.add_subplot(spec[0, 1])
+annotate_axes(ax1, 'ax1')
+ax2 = fig.add_subplot(spec[1, 0])
+annotate_axes(ax2, 'ax2')
+ax3 = fig.add_subplot(spec[1, 1])
+annotate_axes(ax3, 'ax3')
+fig.suptitle('Manually added subplots using add_gridspec')
+
+#############################################################################
+# Axes spanning rows or grids in a grid
+# -------------------------------------
+#
+# We can index the *spec* array using `NumPy slice syntax
+# <https://numpy.org/doc/stable/reference/arrays.indexing.html>`_
+# and the new Axes will span the slice:
+
+fig = plt.figure(figsize=(4.5, 3.5), constrained_layout=True)
+spec = fig.add_gridspec(2, 2)
+ax0 = fig.add_subplot(spec[0, :])
+annotate_axes(ax0, 'ax0')
+ax10 = fig.add_subplot(spec[1, 0])
+annotate_axes(ax10, 'ax10')
+ax11 = fig.add_subplot(spec[1, 1])
+annotate_axes(ax11, 'ax11')
+fig.suptitle('Manually added subplots, spanning a column')
+
+###############################################################################
+# Manual Adjustments to a Gridspec Layout
+# ---------------------------------------
+#
+# When a GridSpec is explicitly used, you can adjust the layout
+# parameters of subplots that are created from the GridSpec.  Note this
```

Comment:
We should mention that this is also possible via the `gridspec_kw` of the high-level plotting functions.